tests/testthat/test-lint_package.R

```diff
@@ -0,0 +1,114 @@
+context("Integration tests for `lint_package`")
+
+# When called from inside a package:
+# lint_package(".")
+# .. should give the same results as when called from outside the package
+# with:
+# lint_package(path_to_package)
+
+# Template packages for use in testing are stored in
+# `tests/testthat/dummy_packages/<pkgName>`
+# These packages should not have a .lintr file:  Hardcoding a .lintr in a
+# dummy package throws problems during `R CMD check` (they are flagged as
+# hidden files, but can't be added to RBuildIgnore since they should be
+# available during `R CMD check` tests)
```

Comment:
As a side note: Couldn't we override the `options(lintr.linter_file = "test_dot_lintr")` to overcome R CMD check warnings about hidden files?